Performance 6 (templates)

frankmayer opened the pull request:
[UPDATED (10.6.2017) - see third paragraph]
Performance (and a bit of cleanup) Batch #6

The changes in this batch are all in files under `templates`.
This PR modifies code to be a bit more performant and also does some cleanup.

I have mostly done work on low hanging fruit. There are still other ways of improving **and cleaning up**, but those involve more deep research in the code and probably more drastic changes in order to be implemented.

**[Update 6.1.2017]**
In order to lighten this PR up, I have introduced 4 sub-PRs, which are easier to digest and mostly with very specific changes. In order to continue with this one, those ones should be tested/reviewed/merged first. After that, I can resolve the conflicts and there should finally be only a few changes left.

- [x] 1) Simplify some ternary operations using elvis operator and remove unnecessary parentheses in templates #13244
- [x] 2) Replace unnecessary double quotes in templates #13245
- [x] 3) Type-safe string comparison in site/templates #13309
- [x] 4) More type-safe comparisons in site/templates #13310

**[Update 10.6.2017]**
All Sub-PRs of this PR have now been merged. That means a lot less changes to be reviewed :smile:

Attention potential testers: Those changes should only need code review.

Thank you!

frankmayer committed:
Type-safe string comparisons

frankmayer committed:
More type-safe comparisons

frankmayer committed:
Replace is_null with ===

frankmayer committed:
Use Elvis instead of complete ternary

frankmayer committed:
Merge nested if

frankmayer committed:
Replace single quotes with double quotes

frankmayer committed:
Remove unnecessary parentheses

frankmayer committed:
Optimize non-optimal if conditions.

frankmayer committed:
Reverted some condition flips as requested in changes of other PR (https://github.com/joomla/joomla-cms/pull/12225)

frankmayer committed:
Reverted some more condition flips as requested in changes of other PR (https://github.com/joomla/joomla-cms/pull/12225)

frankmayer committed:
Reverted even more condition flips as requested in changes of other PR (https://github.com/joomla/joomla-cms/pull/12225)

frankmayer committed:
Reverted one  more condition flips as requested in changes of other PR (https://github.com/joomla/joomla-cms/pull/12225)

frankmayer committed:
Replaced and, or with &&, ||

Bakual commented on templates/beez3/html/com_content/archive/default_items.php:
```diff
@@ -34,7 +34,7 @@
 
 
-<?php if (($params->get('show_author')) or ($params->get('show_parent_category')) or ($params->get('show_category')) or ($params->get('show_create_date')) or ($params->get('show_modify_date')) or ($params->get('show_publish_date'))  or ($params->get('show_hits'))) : ?>
+<?php if ($params->get('show_author') || $params->get('show_parent_category') || $params->get('show_category') || $params->get('show_create_date') || $params->get('show_modify_date') || $params->get('show_publish_date') || $params->get('show_hits')) : ?>
```
In template files (everything within the /tmpl/ folders) we prefer `or` and `and` over `||`  and `&&`. So please leave those alone.

frankmayer replied:
Didn't know that. Why is this?

Bakual replied:
It's easier to read for people (users) which aren't that familiar with PHP.
Same for the `if (...) : ... endif` instead of `if (...) { ... }`

mbabker replied:
When did that become a thing?  Because last I remember that's NOT in our codestyle rules or documented guidelines.  Not to mention there are (mostly small) behavioral differences between `$foo and $bar` versus `$foo && $bar`.

frankmayer replied:
Is there some kind of system with those? I ask because there are also plenty of && and || in those files.
A quick count revealed
84 x `&&` | 29 x `and`
49 x `||` vs x 48 `or`

Bakual replied:
> When did that become a thing? Because last I remember that's NOT in our codestyle rules or documented guidelines. 

Agreed, it's not in the codestyle rules. But I was under the impression that it is usally done like this and I always followed that unspoken rule. Same as the `if ... endif` stuff which isn't (imho) written in the codestyle rules as well.

brianteeman commented:
I hadn't heard that either but I can see the logic behind it.

mbabker commented:
Definitely easier to read but it's not just a 1-for-1 swap between `and` and `&&` or `||` and `or`.

frankmayer commented:
@mbabker could you elaborate on that?

mbabker commented:
http://php.net/manual/en/language.operators.logical.php will probably cover it better than I can.

frankmayer commented:
@mbabker Got it. Thanks.

frankmayer commented:
@mbabker ok but from what I see, most - if not all of those are swap-able 1-for-1. Why is it preferred then? And why specifically for the tmpl?

frankmayer committed:
Comply to change request in PR discussion:  Leave 'or' and 'and' alone

Bakual commented:
The tmpl folder contains the HTML output, which is the stuff users likely want to adjust.
Those files also are easy to override in a template. The files you got in the /html/ folder are actually overrides of the layout files in the /tmpl/ folder.

> most - if not all of those are swap-able 1-for-1

In the layout files, they usually are interchangeable. It only starts to matter if you have them mixed in a assignments and ternaries (see http://php.net/manual/en/language.operators.precedence.php). But there I prefer to use `()` around the logical blocks anyway to make it simpler to read.

frankmayer commented:
Ok, I see, but still it's a mix of both styles, with the `&&` and `||` having the most occurrences. That can be even more confusing to "users".
I would suggest to choose one way and go with it.

Bakual commented:
> I would suggest to choose one way and go with it.

Agreed, but please don't do it in this PR. If we want to choose one way, the proper way is to define a codestyle rule and then apply it. Otherwise you're going to change it now and the next guy wants to change it back 😄

frankmayer commented:
:smile: Yes, of course! I have already reversed those changes.

frankmayer committed:
Missed a spot...

andrepereiradasilva commented on templates/beez3/html/com_content/archive/default_items.php:
```diff
@@ -79,7 +79,7 @@
 	<dd class="createdby">
 		<?php $author = $item->author; ?>
-		<?php $author = ($item->created_by_alias ? $item->created_by_alias : $author);?>
+		<?php $author = ($item->created_by_alias ?: $author);?>
```
like the other PR just
```php
<?php $author = $item->created_by_alias ?: $item->author; ?>
```

andrepereiradasilva commented on templates/beez3/html/com_content/article/default.php:
```diff
@@ -126,7 +126,7 @@
 	<dd class="createdby">
 		<?php $author = $this->item->author; ?>
-		<?php $author = ($this->item->created_by_alias ? $this->item->created_by_alias : $author);?>
+		<?php $author = ($this->item->created_by_alias ?: $author);?>
```
again, just
```php
<?php $author = $this->item->created_by_alias ?: $this->item->author; ?>
```

andrepereiradasilva commented on templates/beez3/html/com_content/category/blog.php:
```diff
@@ -66,14 +66,14 @@
 <?php endif; ?>
 <?php
-	$introcount = (count($this->intro_items));
+	$introcount = count($this->intro_items);
```
actually this seems to be only used once ahead so it should only be there.

see ahead
```php
<?php if (($rowcount == $this->columns) or ($counter == $introcount)) : ?>
```

andrepereiradasilva commented on templates/beez3/html/com_content/category/blog_item.php:
```diff
@@ -104,7 +104,7 @@
 	<dd class="createdby">
 		<?php $author = $this->item->author; ?>
-		<?php $author = ($this->item->created_by_alias ? $this->item->created_by_alias : $author);?>
+		<?php $author = ($this->item->created_by_alias ?: $author);?>
```
again, just
```php
<?php $author = $this->item->created_by_alias ?: $this->item->author; ?>
```

andrepereiradasilva commented on templates/beez3/html/com_content/category/default_articles.php:
```diff
@@ -128,7 +128,7 @@
 						<?php if (!empty($article->author) || !empty($article->created_by_alias)) : ?>
 							<?php $author = $article->author ?>
-							<?php $author = ($article->created_by_alias ? $article->created_by_alias : $author); ?>
+							<?php $author = ($article->created_by_alias ?: $author); ?>
```
again, just
```php
<?php $author = $article->created_by_alias ?: $article->author; ?>
```

andrepereiradasilva commented on templates/beez3/html/com_content/featured/default_item.php:
```diff
@@ -108,7 +108,7 @@
 	<dd class="createdby">
 		<?php $author = $this->item->author; ?>
-		<?php $author = ($this->item->created_by_alias ? $this->item->created_by_alias : $author);?>
+		<?php $author = ($this->item->created_by_alias ?: $author);?>
```
again, just
```php
<?php $author = $this->item->created_by_alias ?: $this->item->author; ?>
```

andrepereiradasilva commented on templates/beez3/jsstrings.php:
```diff
@@ -24,8 +24,8 @@
-	var bildauf    = '" . $this->baseurl . "/templates/" . $this->template . "/images/plus.png';
-	var bildzu     = '" . $this->baseurl . "/templates/" . $this->template . "/images/minus.png';
+	var bildauf    = '" . $this->baseurl . '/templates/' . $this->template . "/images/plus.png';
+	var bildzu     = '" . $this->baseurl . '/templates/' . $this->template . "/images/minus.png';
```
dont't understand why this change, before the baseurl var you have `"` and after `'`. IMHO or we use always `'` or we use always `"`

frankmayer replied:
Thanks, this was an automated change which I obviously reviewed poorly... :smile:

frankmayer replied:
Actually this was supposed to be a HEREDOC.. Forgot to implement...

andrepereiradasilva replied:
@frankmayer please review this

frankmayer replied:
:smile_cat:

andrepereiradasilva commented on templates/beez3/index.php:
```diff
@@ -58,15 +58,16 @@
-	}");
+		background: " . $this->params->get('backgroundcolor') . ';
+	}'
+	);
```
dont't understand why this change, before you have `"` and after `'`. IMHO or we use always `'` or we use always `"`

frankmayer committed:
This way, this JS is much easier to read...

frankmayer committed:
Another HEREDOC

frankmayer committed:
Changes that were pointed out in PR discussion

frankmayer committed:
Hopefully this fixes CS...

andrepereiradasilva commented on templates/beez3/html/com_content/archive/default_items.php:
```diff
@@ -78,9 +78,7 @@
-		<?php $author = $item->author; ?>
-		<?php $author = ($item->created_by_alias ? $item->created_by_alias : $author);?>
-			<?php if (!empty($item->contact_link ) &&  $params->get('link_author') == true):?>
+		<?php $author = $item->created_by_alias ?: $item->author; ?>			<?php if (!empty($item->contact_link ) &&  $params->get('link_author') == true):?>
```
a return is missing here

andrepereiradasilva commented on templates/beez3/html/com_content/article/default.php:
```diff
@@ -125,9 +125,7 @@
-		<?php $author = $this->item->author; ?>
-		<?php $author = ($this->item->created_by_alias ? $this->item->created_by_alias : $author);?>
-		<?php if (!empty($this->item->contact_link ) &&  $params->get('link_author') == true) : ?>
+		<?php $author = $this->item->created_by_alias ?: $this->item->author; ?>		<?php if (!empty($this->item->contact_link ) &&  $params->get('link_author') == true) : ?>
```
a return is missing here

frankmayer committed:
Formatting... ARGH!

andrepereiradasilva commented:
in this one i'm just not sure of the mantainers position in using the JS/CSS HEREDOC for inline js/css in the beez3 template files. I'm not in favour of it, but that is just me.

for the rest it seems ok

frankmayer commented:
No issues, it's just a proposal that I thought of, when I saw it. I can remove it, if it's not wanted. 
However, IMHO it looks cleaner that way, as the variables are defined and the HEREDOC is easier to read and modify if needed.

frankmayer committed:
Merge branch 'staging' into Performance_6

# Conflicts:
#	templates/beez3/component.php
#	templates/protostar/error.php
#	templates/protostar/index.php
#	templates/protostar/offline.php
#	templates/system/error.php
#	templates/system/offline.php

andrepereiradasilva commented:
again, IMHO, the usage of HEREDOC in CSS and JS is a thing that needs some Code Style mantainers decision. IMHO it should be decided if the code should use HEREDOC in all css/js across the code or not.
And that change would be another PR, not this one. But that's my opinion. 

Without the HEREDOC part, as said before, for me is ok on code review.

frankmayer commented:
@andrepereiradasilva I will remove the heredoc changes from this PR, then, so it can move on.

frankmayer committed:
Reverted HEREDOC modifications. (maybe in some other PR?)

andrepereiradasilva commented:
see comment https://github.com/joomla/joomla-cms/pull/12233#discussion_r81654701

frankmayer committed:
Reverted a prior change, which was on hold, as no decision was made up to now, on if we could use HEREDOC to construct JS and CSS instead.

andrepereiradasilva commented:
I have tested this item :white_check_mark: successfully on 1490fabbaf4d08fc7591b4c1a3ef43ffb893e31d<br /><br />code review<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/12233">issues.joomla.org/tracker/joomla-cms/12233</a>.</sub>

frankmayer committed:
Merge branch 'staging' into Performance_6

frankmayer committed:
Merge branch 'staging' into Performance_6

# Conflicts:
#	templates/beez3/html/com_contact/contact/default_form.php
#	templates/beez3/html/com_contact/contact/default_profile.php
#	templates/beez3/html/com_content/archive/default_items.php
#	templates/beez3/html/com_content/article/default.php
#	templates/beez3/html/com_content/category/blog.php
#	templates/beez3/html/com_content/category/blog_item.php
#	templates/beez3/html/com_content/category/default_articles.php
#	templates/beez3/html/com_content/featured/default_item.php
#	templates/beez3/html/modules.php
#	templates/protostar/error.php
#	templates/protostar/index.php

frankmayer committed:
Minor change

zero-24 commented on templates/beez3/html/com_content/category/blog.php:
```diff
@@ -65,11 +65,11 @@
+	<?php
+	$introcount = count($this->intro_items);
+	$counter = 0;
+	?>
```
Please use ;)


```php
<?php $introcount = count($this->intro_items); ?>
<?php $counter = 0; ?>
```

frankmayer committed:
Template code-style fix

frankmayer commented:
Edited initial post to show which sub-PRs still have to go through, in order to make this one easier to digest.
It would be great if the initial big batch PRs of mine could make it into alpha2. Thanks!

frankmayer committed:
Merge branch 'staging' into Performance_6

# Conflicts:
#	templates/beez3/html/com_content/category/blog_item.php
#	templates/beez3/html/com_content/featured/default_item.php
#	templates/protostar/error.php
#	templates/protostar/index.php

frankmayer committed:
Replace unneeded strlen() call with cheaper is-not-empty-string check

frankmayer committed:
Merge branch 'staging' into Performance_6

# Conflicts:
#	templates/beez3/html/com_contact/category/default.php
#	templates/beez3/html/com_newsfeeds/category/default.php
#	templates/beez3/html/com_weblinks/category/default.php
#	templates/beez3/html/mod_breadcrumbs/default.php

frankmayer commented:
[Update 10.6.2017]
All Sub-PRs of this PR have now been merged. That means a lot less changes to be reviewed :smile:

Attention potential testers: Those changes should only need code review.

Thank you!

frankmayer committed:
Changes:
- Replaced AND's and OR's with && and ||
- Modified alias function to use correct function
- Optimization: Extracted variable from multiple calls to do only one call
- Rearranged some conditions in if statements, which cost less than the ones on their left. Logic is still the same.

frankmayer committed:
Reversed AND OR to && || as per discussion in early comments of this PR..

Quy commented on templates/protostar/error.php:
```diff
@@ -25,7 +25,7 @@
 $sitename = $app->get('sitename');
 
-if ($task ==='edit' || $layout === 'form')
+if ($task == ='edit' || $layout === 'form')
```
Fix `== =`

frankmayer replied:
Thanks, hmm must have accidentally hit space while moving the cursor...

Quy commented on templates/protostar/index.php:
```diff
@@ -28,7 +28,7 @@
 $sitename = $app->get('sitename');
 
-if ($task === 'edit' || $layout === 'form')
+if($task === 'edit' || $layout === 'form')
```
Add space after `if`

frankmayer committed:
Corrections according to reviewer's comment

Quy commented on templates/protostar/error.php:
```diff
@@ -25,7 +25,7 @@
 $sitename = $app->get('sitename');
 
-if ($task == ='edit' || $layout === 'form')
+if ($task ==='edit' || $layout === 'form')
```
Add space after `===`

frankmayer committed:
Corrections according to reviewer's comment

Quy commented on templates/beez3/html/com_content/article/default.php:
```diff
@@ -49,16 +54,18 @@
-<?php if ($params->get('access-edit') ||  $params->get('show_print_icon') || $params->get('show_email_icon')) : ?>
+<?php
+
+if ($accessEdit ||  $showPrintIcon || $showEmailIcon) : ?>
```
Remove extra space `||  $showPrintIcon`

frankmayer committed:
Corrections according to reviewer's comment

Quy commented on templates/beez3/html/com_content/article/default.php:
```diff
@@ -9,20 +9,25 @@
 ?>
 <article class="item-page<?php echo $this->pageclass_sfx?>">
-<?php if ($this->params->get('show_page_heading')) : ?>
+<?php if ($showPageHeading = $this->params->get('show_page_heading')) : ?>
```
Substitute with `$showPageHeading` in 2 instances below.

Quy commented on templates/beez3/html/com_content/category/blog.php:
```diff
@@ -108,7 +105,7 @@
 <?php endif; ?>
 
-<?php if (($this->params->def('show_pagination', 1) == 1  || ($this->params->get('show_pagination') == 2)) && ($this->pagination->pagesTotal > 1)) : ?>
+<?php if (($this->pagination->pagesTotal > 1) && ($this->params->def('show_pagination', 1) == 1  || ($this->params->get('show_pagination') == 2))) : ?>
```
Change to. Please review.
```
<?php if ($this->pagination->pagesTotal > 1 && ($this->params->def('show_pagination', 1) == 1 || $this->params->get('show_pagination') == 2)) : ?>
```

Quy commented on templates/beez3/html/com_content/category/blog_item.php:
```diff
@@ -102,8 +102,7 @@
 	<dd class="createdby">
-		<?php $author = $this->item->author; ?>
-		<?php $author = ($this->item->created_by_alias ?: $author);?>
+		<?php $author = $this->item->created_by_alias ?: $this->item->author; ?>
```
Remove extra space `&&  ` below this line.

Quy commented on templates/beez3/html/com_content/category/blog_item.php:
```diff
@@ -161,7 +160,7 @@
 
 <?php if ($this->item->state == 0 || strtotime($this->item->publish_up) > strtotime(JFactory::getDate())
-	|| ((strtotime($this->item->publish_down) < strtotime(JFactory::getDate())) && $this->item->publish_down != JFactory::getDbo()->getNullDate())) : ?>
+	|| ((strtotime($this->item->publish_down) < strtotime(JFactory::getDate())) && $this->item->publish_down !== JFactory::getDbo()->getNullDate())) : ?>
```
Remove outside parentheses.
`(strtotime($this->item->publish_down) < strtotime(JFactory::getDate()))`

Quy commented on templates/beez3/html/com_content/category/default.php:
```diff
@@ -9,14 +9,16 @@
-$pageClass = $this->params->get('pageclass_sfx');
+$showCategoryTitle            = $this->params->get('show_category_title') == 1;
+$showCategoryHeadingTitleText = $this->params->get('show_category_heading_title_text', 1) == 1;
+$pageClass                    = $this->params->get('pageclass_sfx');
```
`$pageClass` not utilized.

frankmayer replied:
Nice catch

Quy commented on templates/beez3/html/com_content/category/default.php:
```diff
@@ -9,14 +9,16 @@
 JHtml::_('behavior.caption');
 
-$pageClass = $this->params->get('pageclass_sfx');
+$showCategoryTitle            = $this->params->get('show_category_title') == 1;
```
There are several instances below that can be replaced with this variable.

frankmayer replied:
I think those things went down the drain when I reverted the `and` and `or` stuff. BTW, if you are interested,  I'd like your opinion on the matter of my proposal to replace those `and` and `or` that I made in the Joomla code-styling repository.
Thanks for reviewing, again!

frankmayer replied:
Done

Quy commented on templates/beez3/html/com_content/category/default.php:
```diff
@@ -47,33 +49,33 @@
 
 
-<?php if (is_array($this->children[$this->category->id]) && count($this->children[$this->category->id]) > 0 && $this->params->get('maxLevel') != 0) : ?>
+<?php if ($this->params->get('maxLevel') != 0 && (is_array($this->children[$this->category->id]) && count($this->children[$this->category->id]) > 0)) : ?>
```
Remove outside parentheses.
`(is_array($this->children[$this->category->id]) && count($this->children[$this->category->id]) > 0)`

Quy commented on templates/beez3/html/com_content/category/default_articles.php:
```diff
@@ -154,8 +154,7 @@
 						<?php if (!empty($article->author) || !empty($article->created_by_alias)) : ?>
-							<?php $author = $article->author ?>
-							<?php $author = ($article->created_by_alias ?: $author); ?>
+							<?php $author = $article->created_by_alias ?: $article->author; ?>
```
Remove extra space below `&&  `

Quy commented on templates/beez3/html/com_content/featured/default_item.php:
```diff
@@ -11,15 +11,19 @@
 
 <?php if ($this->item->state == 0 || strtotime($this->item->publish_up) > strtotime(JFactory::getDate())
-	|| ((strtotime($this->item->publish_down) < strtotime(JFactory::getDate())) && $this->item->publish_down != JFactory::getDbo()->getNullDate())) : ?>
+	|| ((strtotime($this->item->publish_down) < strtotime(JFactory::getDate())) && $this->item->publish_down !== JFactory::getDbo()->getNullDate())) : ?>
```
Remove outside parentheses.
`(strtotime($this->item->publish_down) < strtotime(JFactory::getDate()))`

Quy commented on templates/beez3/html/com_content/featured/default_item.php:
```diff
@@ -167,7 +170,7 @@
 
 <?php if ($this->item->state == 0 || strtotime($this->item->publish_up) > strtotime(JFactory::getDate())
-	|| ((strtotime($this->item->publish_down) < strtotime(JFactory::getDate())) && $this->item->publish_down != JFactory::getDbo()->getNullDate())) : ?>
+	|| ((strtotime($this->item->publish_down) < strtotime(JFactory::getDate())) && $this->item->publish_down !== JFactory::getDbo()->getNullDate())) : ?>
```
Remove outside parentheses.
`(strtotime($this->item->publish_down) < strtotime(JFactory::getDate()))`

Quy commented on templates/beez3/html/mod_breadcrumbs/default.php:
```diff
@@ -47,7 +47,7 @@
 		}
 
-		if (($key != $penult_item_key) || $show_last)
+		if (($key !== $penult_item_key) || $show_last)
```
Remove parentheses

Quy commented on templates/protostar/index.php:
```diff
@@ -92,15 +92,18 @@
 // Adjusting content width
-if ($this->countModules('position-7') && $this->countModules('position-8'))
+$position8ModuleCount = $this->countModules('position-8');
+$position7ModuleCount = $this->countModules('position-7');
```
Trivial change. Swap lines (7 before 8).

frankmayer replied:
Ah, yes forgot that... Will address all your reviews and push a new commit. Thanks

frankmayer committed:
Some more changes + changes according to reviewer's comments

Quy commented on templates/beez3/html/com_content/category/blog.php:
```diff
@@ -105,7 +105,7 @@
 <?php endif; ?>
 
-<?php if (($this->pagination->pagesTotal > 1) && ($this->params->def('show_pagination', 1) == 1  || ($this->params->get('show_pagination') == 2))) : ?>
+<?php if ($this->pagination->pagesTotal > 1 && ($this->params->def('show_pagination', 1) == 1  || ($this->params->get('show_pagination') == 2))) : ?>
```
Remove extra space and outside parentheses.
`  || ($this->params->`

Quy commented on templates/beez3/html/com_content/category/blog_item.php:
```diff
@@ -103,7 +103,7 @@
 	<dd class="createdby">
 		<?php $author = $this->item->created_by_alias ?: $this->item->author; ?>
-		<?php if (!empty($this->item->contact_link ) &&  $params->get('link_author') == true) : ?>
+		<?php if (!empty($this->item->contact_link ) && $params->get('link_author') == true) : ?>
```
Remove extra space `item->contact_link )`

frankmayer committed:
Changes according to reviewer's comments

Quy commented on templates/beez3/html/com_content/article/default.php:
```diff
@@ -123,10 +130,9 @@
 		</dd>
 <?php endif; ?>
-<?php if ($params->get('show_author') && !empty($this->item->author )) : ?>
+<?php if (!empty($this->item->author ) && $params->get('show_author')) : ?>
```
Remove extra space `($this->item->author )`

Quy commented on templates/beez3/html/com_content/article/default.php:
```diff
@@ -123,10 +130,9 @@
 	<dd class="createdby">
-		<?php $author = $this->item->author; ?>
-		<?php $author = ($this->item->created_by_alias ?: $author);?>
+		<?php $author = $this->item->created_by_alias ?: $this->item->author; ?>
```
Remove extra space `($this->item->contact_link )`

Quy commented on templates/beez3/html/com_content/category/blog.php:
```diff
@@ -84,7 +81,7 @@
 		</article>
 		<?php $counter++; ?>
-		<?php if (($rowcount == $this->columns) or ($counter == $introcount)) : ?>
+		<?php if (($rowcount === $this->columns) or ($counter === $introcount)) : ?>
```
Remove parentheses.

Quy commented on templates/beez3/html/com_content/category/blog.php:
```diff
@@ -96,7 +93,7 @@
 <?php endif; ?>
 
-<?php if (is_array($this->children[$this->category->id]) && count($this->children[$this->category->id]) > 0 && $this->params->get('maxLevel') != 0) : ?>
+<?php if ($this->params->get('maxLevel') != 0 && (is_array($this->children[$this->category->id]) && count($this->children[$this->category->id]) > 0)) : ?>
```
Remove parentheses.

Quy commented on templates/beez3/html/com_content/category/blog_item.php:
```diff
@@ -18,7 +18,7 @@
 
 <?php if ($this->item->state == 0 || strtotime($this->item->publish_up) > strtotime(JFactory::getDate())
-	|| ((strtotime($this->item->publish_down) < strtotime(JFactory::getDate())) && $this->item->publish_down != JFactory::getDbo()->getNullDate())) : ?>
+	|| ((strtotime($this->item->publish_down) < strtotime(JFactory::getDate())) && $this->item->publish_down !== JFactory::getDbo()->getNullDate())) : ?>
```
Remove parentheses.

Quy commented on templates/beez3/html/com_content/category/default.php:
```diff
@@ -9,35 +9,38 @@
-<?php if ($this->params->get('show_page_heading') and ($this->params->get('show_category_title') or $this->params->get('page_subheading'))) : ?>
+<?php
+if ($this->params->get('show_page_heading')) : ?>
+<?php if ($this->params->get('show_page_heading') and ($showCategoryTitle === true or $pageSubHeading)) : ?>
```
Redundant checking with the previous line `$this->params->get('show_page_heading'`.

frankmayer committed:
Merge branch 'staging' into Performance_6

frankmayer committed:
Changes according to reviewer's comments

andrepereiradasilva commented on templates/beez3/html/com_content/article/default.php:
```diff
@@ -42,23 +47,25 @@
 		</h2>
 <?php endif; ?>
-<?php if ($this->params->get('show_page_heading') and $params->get('show_title')) :?>
+<?php if ($showPageHeading and $params->get('show_title')) :?>
```
are you sure `$showPageHeading` defined at this time?

frankmayer replied:
It is defined in the first if() statement at line 30. Or, did I miss something?

andrepereiradasilva replied:
right sorry ... tought this instruction was inside an `else`

frankmayer replied:
No issue, thought I had messed up :)

andrepereiradasilva commented on templates/beez3/html/com_content/category/blog.php:
```diff
@@ -84,7 +81,7 @@
 		</article>
 		<?php $counter++; ?>
-		<?php if (($rowcount == $this->columns) or ($counter == $introcount)) : ?>
+		<?php if ($rowcount === $this->columns or $counter === $introcount) : ?>
```
are you sure `$this->columns` is an integer ? (see line above `echo (int) $this->columns;`)

frankmayer replied:
Thank you. Somehow got that wrong...

andrepereiradasilva reviewed (commented):
two doubts

frankmayer committed:
Changes according to reviewer's comment

Quy commented on templates/beez3/html/com_content/article/default.php:
```diff
@@ -123,11 +130,10 @@
-		<?php $author = ($this->item->created_by_alias ?: $author);?>
-		<?php if (!empty($this->item->contact_link ) &&  $params->get('link_author') == true) : ?>
+		<?php $author = $this->item->created_by_alias ?: $this->item->author; ?>
+		<?php if (!empty($this->item->contact_link) &&  $params->get('link_author') == true) : ?>
```
Remove extra space

frankmayer committed:
Changes according to reviewer's comment

andrepereiradasilva commented:
I have tested this item :white_check_mark: successfully on 4f3c91f8405f42b6fe4235ea19a02794ad5ec2bb<br /><br />code review<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/12233">issues.joomla.org/tracker/joomla-cms/12233</a>.</sub>

Quy commented:
I have tested this item :white_check_mark: successfully on 4f3c91f8405f42b6fe4235ea19a02794ad5ec2bb<br /><br />Code review.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/12233">issues.joomla.org/tracker/joomla-cms/12233</a>.</sub>

frankmayer commented:
RTC please, thank you

franz-wohlkoenig commented:
RTC after two successful tests.

yvesh commented on templates/beez3/html/com_content/article/default.php:
```diff
@@ -9,22 +9,27 @@
+<?php if ($showPageHeading = $this->params->get('show_page_heading')) : ?>
 
-<?php if ($this->params->get('show_page_heading') and $params->get('show_title')) :?>
+<?php if ($showPageHeading and $params->get('show_title')) :?>
```
Is there any reason we are using `and` here? :-)

frankmayer replied:
Since this really big one is already RTC, and those seem to be no show-stoppers, do you mind marking those reviews as non-stoppers? It took a long time to go RTC for those bug ones :). 
I will happily open a new PR for those ones after this is merged.

Also for this one and the one after the next, those are mixed files, and there has been lot of discussion on that in one of my other PR's.. in the code-standards Repo. Still ongoing I think.

yvesh commented on templates/beez3/html/com_content/category/blog_item.php:
```diff
@@ -161,7 +160,7 @@
 
 <?php if ($this->item->state == 0 || strtotime($this->item->publish_up) > strtotime(JFactory::getDate())
-	|| ((strtotime($this->item->publish_down) < strtotime(JFactory::getDate())) && $this->item->publish_down != JFactory::getDbo()->getNullDate())) : ?>
+	|| (strtotime($this->item->publish_down) < strtotime(JFactory::getDate()) && $this->item->publish_down !== JFactory::getDbo()->getNullDate())) : ?>
```
instead of calling strtotime on JFactory::getDate you can use JFactory::getDate()->getTimestamp();

yvesh commented on templates/beez3/html/com_content/category/default.php:
```diff
@@ -9,35 +9,38 @@
 <?php endif; ?>
 
-<?php if ($this->params->get('show_category_title') or $this->params->get('page_subheading')) : ?>
+<?php if ($showCategoryTitle === true or $pageSubHeading) : ?>
```
Why are we using or here? :-)

yvesh commented on templates/beez3/html/com_newsfeeds/category/default_children.php:
```diff
@@ -10,12 +10,12 @@
 <?php foreach ($this->children[$this->category->id] as $id => $child) : ?>
 	<?php
-	if ($this->params->get('show_empty_categories') || $child->numitems || count($child->getChildren())) :
+	if ($child->numitems || $this->params->get('show_empty_categories') || count($child->getChildren())) :
```
empty is way faster count iirc

yvesh commented on templates/beez3/html/mod_breadcrumbs/default.php:
```diff
@@ -35,7 +35,7 @@
 	// Make a link if not the last item in the breadcrumbs
 	$show_last = $params->get('showLast', 1);
-	if ($key != $last_item_key)
+	if ($key !== $last_item_key)
```
new line missing before (yep i know it has been like that

rdeutz commented:
I am not going to merge these types of PR into 3.7.4, we have seen that chances are good to miss something and then we introduce a bug as we have seen in the last release. Make the software more solid and stable is more important.

frankmayer commented:
@mbabker should I change the base of this PR and of https://github.com/joomla/joomla-cms/pull/12220 to 3.8-dev and resolve any conflicts if there are any?

mbabker commented:
That'd be helpful.

frankmayer commented:
Base changed to 3.8-dev without conflicts on this PR.

frankmayer commented:
@mbabker Base changed to 3.8-dev on this PR. Conflicts resolved.
